src/service_manager.rs

```diff
@@ -208,4 +209,41 @@ impl ServiceManager {
             Ok(Service::new(unsafe { ScHandle::new(service_handle) }))
         }
     }
+
+    /// Retrieves the service name of the specified service.
+    pub fn get_service_key_name(&self, display_name: impl AsRef<OsStr>) -> Result<OsString> {
+        let service_display_name =
+            WideCString::from_os_str(display_name).map_err(|_| Error::DisplayNameHasNulByte)?;
+        let mut size = 0u32;
+        let res = unsafe {
+            Services::GetServiceKeyNameW(
+                self.manager_handle.raw_handle(),
+                service_display_name.as_ptr(),
+                std::ptr::null_mut(),
+                &mut size as _,
```

Comment:
Instead of probing for the buffer size we usually just allocate the maximum buffer size directly and get away with one syscall. Because a small allocation is definitely faster than a syscall. Unless there is some syscall fast path here I'm not aware of.

See this method for reference: https://github.com/mullvad/windows-service-rs/blob/24bd91a84529a315277b7096a52d67057a394edb/src/service.rs#L1465-L1487

According to [the docs](https://docs.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-getservicekeynamew) the max size should be 4k.

What do you think about this approach here, and in general?

Reply:
I'm used to allocate buffers of the needed space, but I guess this approach works fine (as long as the max size is reasonable for a default value).

Done